entity/src/content_audit.rs

```diff
@@ -335,6 +376,56 @@ pub async fn get_audits<T: OverlayContentKey>(
         .await?)
 }
 
+pub async fn get_failed_keys(
+    subprotocol: SubProtocol,
+    strategy_used: String,
+    page: u32,
+    conn: &DatabaseConnection,
+) -> Result<Vec<String>> {
+    let page_size: u32 = 1000;
```

Comment:
small semantic thing, but it's good to define constants as constants:
```suggestion
    const PAGE_SIZE: u32 = 1000;
```